course_info/views.py

```diff
@@ -98,79 +122,34 @@ def __course_context(request, keys, show_empty_field=False, **kwargs):
-        :return: The formatted field value
-        """
-        if value:
-            if label and show_label_if_not_empty:
-                value = label + value
-        elif show_empty_field:
-            value = 'Field not populated by registrar'
-            if label:
-                value = label + value
-
-        return value
-
-    for field in fields:
-        if field['key'] == 'title':
-            field['value'] = "<h4>%s</h4>" % field_value_display(field['value'], "Course Title: ", True, False)
-        elif field['key'] == 'term.display_name':
-            field['value'] = field_value_display(field['value'], "<b>Term:</b> ", show_empty_field, False)
-        elif field['key'] == 'instructors_display':
-            field['value'] = field_value_display(field['value'], "<b>Instructors:</b> ", show_empty_field, False)
-        elif field['key'] == 'location':
-            field['value'] = field_value_display(field['value'], "<b>Location:</b> ", show_empty_field)
-        elif field['key'] == 'meeting_time':
-            field['value'] = field_value_display(field['value'], "<b>Meeting Time:</b> ", show_empty_field)
-        elif field['key'] == 'exam_group':
-            field['value'] = field_value_display(field['value'], "<b>Exam Group:</b> ", show_empty_field)
-        elif field['key'] == 'description':
-            field['value'] = field_value_display(field['value'], "<b>Course Description:</b> ", show_empty_field)
-        elif field['key'] == 'notes':
-            field['value'] = field_value_display(field['value'], "<b>Notes:</b> ", show_empty_field)
-        elif field['key'] == 'course.registrar_code_display':
-            try:
-                # Extracts the course number (the last substring) from typical display codes
-                field['value'] = field_value_display(field['value'].split()[-1], school_name + ": ", show_empty_field)
-            except IndexError:
-                # If the display code is atypical, show the whole thing
-                field['value'] = field_value_display(field['value'], school_name + ": ", show_empty_field)
-        field['value'] = field['value'].replace('<br /> <br />', '<br />')
-
-    return fields
+    return context
 
 
 @require_GET
 def widget(request):
+    """
+    Returns course information (the results of an iCommons API query), called
+    from an embedded iframe in Canvas. Note that this is NOT a secure endpoint,
+    and spoofing HTTP_REFERER makes this information available for any course,
+    so if anything is added of a sensitive nature then this may need a different
+    implementation.
+    """
     referer = request.META.get('HTTP_REFERER', '')
     try:
         canvas_course_id = re.match('^.+/courses/(?P<canvas_course_id>\d+)(?:$|.+$)', referer).group('canvas_course_id')
     except AttributeError:
         canvas_course_id = None
 
+    # field names are sent as URL params f=field_name when widget is 'launched'
     field_names = request.GET.getlist('f')
```

Comment:
does this potentially also return a `None` as a field name, if `&f=` is in the params?